src/index.ts

```diff
@@ -119,17 +119,67 @@ function makePassesFilter(opts: FilterOpts): (msg: Msg) => boolean {
 class threads {
   private readonly ssb: Record<string, any>;
   private readonly isBlocking: (obj: any, cb: CB<boolean>) => void;
+  private readonly isFollowing: ((obj: any, cb: CB<boolean>) => void) | null;
 
   constructor(ssb: Record<string, any>, _config: any) {
     this.ssb = ssb;
     this.isBlocking = ssb.friends?.isBlocking
       ? ssb.friends.isBlocking
       : IS_BLOCKING_NEVER;
+    this.isFollowing = ssb.friends?.isFollowing;
     this.ssb.db.registerIndex(HashtagsPlugin);
   }
 
   //#region PRIVATE
 
+  private maybeOnlyKeepFollowingMessages =
+    (enabled: boolean) => (source: any) =>
+      pull(
+        source,
+        pull.asyncMap((msg: Msg, cb: CB<Msg | null>) => {
+          if (!enabled) {
+            cb(null, msg);
+            return;
+          }
+
+          if (!this.isFollowing) {
+            cb(null, msg);
+            return;
+          }
+
+          function isFollowingCb(err: any, following: boolean) {
+            if (err) cb(err);
+            else if (!following) cb(null, null);
+            else cb(null, msg);
+          }
+
+          const rootMsgKey = getRootMsgId(msg);
+
+          if (rootMsgKey === msg.key) {
```

Comment:
I suspect this will be true always, in `public()` and `publicSummary()` (let's talk about `publicUpdates()` later since it's a special case and maybe should be implemented separately). In fact maybe you can `test.skip` the publicUpdates and run `npm run coverage`, I think the coverage results would show that the branch on line 166 is never run.

This is good news because it means you can delete code. i.e. lines 166–178 and you can also assume that line 158 is always true.